File: solutions/remove_element.py
# Problem: 27. Remove Element

def remove_element(nums, val):
    counter = len(nums)
    while val in nums:
        nums.remove(val)
        counter -= 1
    for _ in range(counter):
        nums.append('_')
    return counter


answer = remove_element(nums=[3, 2, 2, 3], val=3)
print(answer)

File: solutions/go/remove_element.go
package main

import (
	"fmt"
)

func removeElement(nums []int, val int) int {
	i := 0
	for i <= len(nums)-1 {
		if nums[i] == val {
			nums = append(nums[:i], nums[i+1:]...)
		} else {
			i++
		}
	}
	return len(nums)
}

func main() {
	result := removeElement([]int{0, 1, 2, 2, 3, 0, 4, 2}, 2)
	fmt.Printf("Result: %v\n", result)
}
